crates/server/src/authority/authority.rs

```diff
@@ -203,3 +203,180 @@ pub trait DnssecAuthority: Authority {
+///   immediately to the client, without consulting any other authorities.  For example, if the
+///   the user configures a blocklist authority, it would not be appropriate to pass the query to
+///   any additional authorities to try to resolve, as that might be used to leak information to a
+///   hostile party, and so a blocklist (or similar) authority should wrap responses for any
+///   blocklist hits in LookupControlFlow::Break.
+/// * Authorities may use LookupControlFlow::Skip to indicate the authority did not attempt to
+///   process a particular query.  This might be used, for example, in a block list authority for
+///   any queries that **did not** match the blocklist, to allow the recursor or forwarder to
+///   resolve the query. Skip must not be used to represent an empty lookup; (use
+///   Continue(EmptyLookup) or Break(EmptyLookup) for that.)
+pub enum LookupControlFlow<T, E = LookupError> {
+    /// A lookup response that may be passed to one or more additional authorities before
+    /// being returned to the client.
+    Continue(Result<T, E>),
+    /// A lookup response that must be immediately returned to the client without consulting
+    /// any other authorities.
+    Break(Result<T, E>),
+    /// The authority did not answer the query and the next authority in the chain should
+    /// be consulted.
+    Skip,
+}
+
+/// The following are a minimal set of methods typically used with Result or Option, and that
+/// were used in the server code or test suite prior to when the LookupControlFlow type was created
+/// (authority lookup functions previously returned a Result over a Lookup or LookupError type.)
+impl<T, E> LookupControlFlow<T, E> {
+    /// Return true if self is LookupControlFlow::Continue
+    pub fn is_continue(&self) -> bool {
+        matches!(self, Self::Continue(_))
+    }
+
+    /// Return true if self is LookupControlFlow::Break
+    pub fn is_break(&self) -> bool {
+        matches!(self, Self::Break(_))
+    }
+    /// Return an &str representation of the current variant of self
+    pub fn variant_as_str(&self) -> &str {
+        match self {
+            Self::Continue(cont) => match cont {
+                Ok(_) => "LookupControlFlow::Continue(Ok)",
+                Err(_) => "LookupControlFlow::Continue(Err)",
+            },
+            Self::Break(b) => match b {
+                Ok(_) => "LookupControlFlow::Break(Ok)",
+                Err(_) => "LookupControlFlow::Break(Err)",
+            },
+            Self::Skip => "LookupControlFlow::Skip",
+        }
+    }
+}
+
+impl<T: LookupObject + 'static, E: std::fmt::Display> LookupControlFlow<T, E> {
+    /// Return inner Ok variant or panic with a custom error message.
+    pub fn expect(self, msg: &str) -> T {
+        match self {
+            Self::Continue(Ok(ok)) | Self::Break(Ok(ok)) => ok,
+            _ => {
+                panic!(
+                    "LookupControlFlow::expect called on unexpected variant {}: {msg}",
```

Comment:
Nit: add `()` suffixes to the method names in these panic messages (so `expect()`)
?